fabric/src/main/java/com/vexsoftware/votifier/fabric/forwarding/FabricMessagingForwardingSink.java

```diff
@@ -0,0 +1,41 @@
+package com.vexsoftware.votifier.fabric.forwarding;
+
+import com.vexsoftware.votifier.fabric.NuVotifier;
+import com.vexsoftware.votifier.support.forwarding.AbstractPluginMessagingForwardingSink;
+import com.vexsoftware.votifier.support.forwarding.ForwardedVoteListener;
+import net.fabricmc.fabric.api.networking.v1.PacketSender;
+import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.server.network.ServerGamePacketListenerImpl;
+
+public class FabricMessagingForwardingSink extends AbstractPluginMessagingForwardingSink implements ServerPlayNetworking.PlayChannelHandler {
+
+    private final String channel;
+
+    public FabricMessagingForwardingSink(String channel, ForwardedVoteListener listener) {
+        super(listener);
+        this.channel = channel;
+        ServerPlayNetworking.registerGlobalReceiver(new ResourceLocation(channel), this);
+    }
+
+    @Override
+    public void halt() {
+        ServerPlayNetworking.unregisterGlobalReceiver(new ResourceLocation(channel));
+    }
+
+    @Override
+    public void receive(MinecraftServer server, ServerPlayer player, ServerGamePacketListenerImpl handler, FriendlyByteBuf buf, PacketSender responseSender) {
+        byte[] data = new byte[buf.capacity()];
+        for (int i = 0; i < buf.capacity(); i++) {
+            data[i] = buf.getByte(i);
+        }
```

Comment:
Why not this?
```suggestion
        byte[] data = new byte[buf.readableBytes()];
        buf.readBytes(data);
```
The loop you're using here is going to be pretty slow.